.github/workflows/check-dist.yml

```diff
@@ -70,3 +73,14 @@ jobs:
         with:
           name: dist
           path: dist/
+
+      - if:
+          ${{ failure() && steps.diff.outcome == 'failure' && github.event_name
+          == 'pull_request'}}
+        name: Push updated build
+        run: |
+          git config --global user.name pulumi-bot
+          git config --global user.email bot@pulumi.com
+          git add dist/
+          git commit -m "Rebuild dist"
+          git push
```

Comment:
It's not clear to me what this is pushing back to? Doesn't this script run only on `main`? I'd rather not set up a CI job that pushes directly to main.

Reply:
This pushes back to the PR's branch - therefore allowing the failing PR to start passing.

Reply:
If it's not running against a PR it'll skip this step.

Reply:
Does this manifest as SDK regeneration? I'm having trouble groking the desired change. `dist` is not meaningful to me.

Reply:
Oh right - nodejs actions have to have compilled code committed to be runnable without extra tooling. This goes into the `dist` folder.

This isn't related to providers, though the same technique could perhaps be applied to providers for re-generating schema & SDKs if it was forgotten to be rebuilt. I think Guin was working on something like this.